internal/bzlmod/go_deps.bzl

```diff
@@ -42,10 +64,27 @@ def _go_deps_impl(module_ctx):
 
     outdated_direct_dep_printer = print
     for module in module_ctx.modules:
-        # Parse the go_dep.config tag of the root module only.
+        def fail_unless_overriding_tolerated(attribute_name, tag_class):
+            if not (module.is_root or _tolerate_overrides_in_module(module)):
+                fail("""\
+Overriding the "{attribute_name}" attribute in a "go_deps.{tag_class}" tag
+is forbidden in non-root Bazel modules, but module "{module_name}" requests it.
+
+If you need this override for a module included in the Bazel Central Registry,
```

Comment:
This already reads great!

Maybe use "to be included" (or some other construction) instead of "included" to make it clear that a module doesn't have to be in the registry (yet) for its Gazelle overrides to be eligible for inclusion in the registry.

Reply:
Ah, I struggled with that part of the sentence, and edited it several times, trying to figure out the idea in your original proposal.

Is the point that we want to encourage contributing overrides for—and extracting them from—modules that are likely to become dependencies of other modules, such that these modules won't be the root module and hence can't host the overrides themselves? In other words, entering the BCR is a sign that you need other modules to be able to make use of you. Registering a root module—one intended to stay as the root—wouldn't make sense.

Reply:
We want to achieve multiple goals with the registry of Gazelle defaults:

* Regular users with root-only projects should not have to set as many Gazelle directives manually.
* Bazel module authors should be able to submit their modules to registries even if they have Go dependencies that require Gazelle directives.

Only the second point seems relevant in the context of this error message, but both are of general importance.